core/domain/platform_parameter_domain.py

```diff
@@ -35,3 +46,935 @@ class PlatformParameterChange(change_domain.BaseChange):
+                'Unsupported data type: %s.' % self._data_type)
+
+        for rule in self._rules:
+            predicate = self.DATA_TYPE_PREDICATES_DICT[self.data_type]
+            if not predicate(rule.value_when_matched):
+                raise utils.ValidationError(
+                    'Expected %s, received %s in value_when_matched' % (
+                        self._data_type, rule.value_when_matched))
+            rule.validate()
+
+        if self._metadata.is_feature:
+            self._validate_feature_flag()
+
+    def evaluate(self, context):
+        """Evaluates the value of the platform parameter in the given context.
+        The value of first matched rule is returned as the result.
+
+        Args:
+            context: EvaluationContext. The context for evaluation.
+
+        Returns:
+            *. The evaluate result of the platform parameter.
+        """
+        for rule in self._rules:
+            matched, value = rule.evaluate(context)
+            if matched:
+                return value
+
+    def to_dict(self):
+        """Returns a dict representation of the PlatformParameter domain
+        object.
+
+        Returns:
+            dict. A dict mapping of all fields of PlatformParameter object.
+        """
+        return {
+            'name': self._name,
+            'description': self._description,
+            'data_type': self._data_type,
+            'rules': [rule.to_dict() for rule in self._rules],
+            'rule_schema_version': self._rule_schema_version,
+            'metadata': self._metadata.to_dict()
+        }
+
+    def _validate_feature_flag(self):
+        """Validates the PlatformParameter domain object that is a feature
+        flag.
+        """
+        if self._data_type != 'bool':
+            raise utils.ValidationError(
+                'Data type of feature flags must be bool, got %s '
+                'instead.' % self._data_type)
+        if self._metadata.stage not in self.ALLOWED_FEATURE_STAGE:
+            raise utils.ValidationError(
+                'Invalid feature stage, expected on of %s, got %s.' % (
+                    self.ALLOWED_FEATURE_STAGE, self._data_type))
+        enabling_rules = [
+            rule for rule in self._rules if rule.value_when_matched]
+        for rule in enabling_rules:
+            if not rule.has_mode_filter():
+                raise utils.ValidationError(
+                    'Rules that enable a feature must have a mode '
+                    'filter.')
+            mode_filters = [
+                mode_filter for mode_filter in rule.filters
+                if mode_filter.type == 'mode']
+            for mode_filter in mode_filters:
+                value_list = (
+                    mode_filter.value if isinstance(mode_filter.value, list)
+                    else [mode_filter.value])
+                if self._metadata.stage == 'dev':
+                    if 'test' in value_list or 'prod' in value_list:
```

Comment:
Change these (dev, prod, test) to constants.

Reply:
Done.